pkg/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -175,33 +175,109 @@ func (r *ReconcileWorkspaceRouting) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	ingressesInSync, err := r.syncIngresses(instance, ingresses)
+	ingressesInSync, clusterIngresses, err := r.syncIngresses(instance, ingresses)
 	if err != nil || !ingressesInSync {
 		reqLogger.Info("Ingresses not in sync")
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	routesInSync, err := r.syncRoutes(instance, routes)
+	routesInSync, clusterRoutes, err := r.syncRoutes(instance, routes)
 	if err != nil || !routesInSync {
 		reqLogger.Info("Routes not in sync")
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	return reconcile.Result{}, r.reconcileStatus(instance, routingObjects)
+	exposedEndpoints, endpointsReady, err := getExposedEndpoints(instance.Spec.Endpoints, clusterIngresses, clusterRoutes)
```

Comment:
`exposedEndpoints, endpointsReady` makes an impression that two lists are returned - all exposed, only ready ones.

`exposedEndpoints, areReady` might be a bit better, might be not. Up to you

Reply:
renamed `endpointsReady` -> `endpointsAreReady`. I prefer to keep `endpoints` in the name as otherwise we just have a variable `areReady` floating around. WDYT?